twitterpuller/src/main/java/de/brandwatch/minianalytics/twitterpuller/twitter/TweetSearcher.java

```diff
@@ -0,0 +1,64 @@
+package de.brandwatch.minianalytics.twitterpuller.twitter;
+
+import de.brandwatch.minianalytics.twitterpuller.kafka.Producer;
+import de.brandwatch.minianalytics.twitterpuller.model.Resource;
+import de.brandwatch.minianalytics.twitterpuller.postgres.cache.QueryCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import twitter4j.*;
+
+import java.time.Instant;
+
+@Component
+public class TweetSearcher {
+
+    private static final Logger logger = LoggerFactory.getLogger(TweetSearcher.class);
+
+    private final QueryCache queryCache;
+
+    private final Producer producer;
+
+    private final Twitter twitter;
+
+    @Autowired
+    public TweetSearcher(QueryCache queryCache, Producer producer, Twitter twitter) {
+        this.queryCache = queryCache;
+        this.producer = producer;
+        this.twitter = twitter;
+    }
+
+
+    @Scheduled(fixedDelay = 1000)
+    public void searchTweets(){
+        queryCache.getCachedQueries().forEach(this::search);
+    }
+
+    private void search(de.brandwatch.minianalytics.twitterpuller.postgres.model.Query query){
+        try {
+            Query searchQuery = new Query(query.getQuery());
+            searchQuery.setCount(100);
+
+            QueryResult queryResult = twitter.search(searchQuery);
+
+            queryResult.getTweets().stream()
+                    .map(this::mapToResource)
+                    .forEach(producer::send);
+
+        } catch (TwitterException e) {
+            logger.warn(e.getErrorMessage());
```

Comment:
Do you have a good reason to not log this as `logger.error`? There could be, I am just wondering.

Also you can maybe add a meaningful message:

```suggestion
            logger.error("Exception occurred while fetching query data from Twitter", e.getErrorMessage());
```

Reply:
Didn't know `logger.error` existed. I will change it